rune/libcontainer/specconv/spec_linux.go

```diff
@@ -202,11 +202,14 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 	// the hint about whether touching with enclave devices or not in container.
 	createEnclaveConfig(spec, config)
 
-	exists := false
+	exists, mountAesmd := false, false
 	for _, m := range spec.Mounts {
 		config.Mounts = append(config.Mounts, createLibcontainerMount(cwd, m))
+		if strings.EqualFold(m.Destination, "/var/run/aesmd") {
+			mountAesmd = true
+		}
 	}
-	if config.Enclave != nil {
+	if config.Enclave != nil && !mountAesmd {
```

Comment:
It is better to write a separate function inside libenclave and called by this function.

```go
func CreateLibenclaveMount(...) {
   for _, m := range spec.Mounts {
       if strings.EqualFold(m.Destination, "/var/run/aesmd") {
           aesmedMounted = true
           break;
       }
   }

   if aesmedMounted == true {
       onfig.Mounts = append(config.Mounts, createLibenclaveMount(cwd))
   }
}
```